### PR TITLE
Feat/phone auth

### PR DESCRIPTION
**Pull request by thelastandrew**
Features implemented:
- phone auth

Features removed:
- email/pass registration
- email/pass auth
- reset pass

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,41 +1,26 @@
...
-import { Home, Register, Login, Dashboard, Profile, NotFound } from 'pages';
-import { useStore } from 'store';
+import { Home, Dashboard, Profile, NotFound, SignIn } from 'pages';
+import { useStore } from 'store/store';
 
 const App: FC = () => {
-  const { t } = useTranslation();
-
-  const { isFetching, getUser } = useStore((state) => state);
+  const { getAuth, isFetching } = useStore((state) => state);
```
Comment (remadyt) [inline src/App.tsx:19]: incorrect hook name, rename to useAuth

File: src/App.tsx [hunk 2]
```diff
@@ -1,41 +1,26 @@
...
-        errorNotification(title, message);
-      }
-    };
-
-    getIsError();
-  }, [getUser, t]);
+    getAuth();
+  }, [getAuth]);
 
   if (isFetching) return <LoaderScreen />;
```
Comment (remadyt) [inline src/App.tsx:25]: create container component for loader

File: src/components/Header/Header.tsx [hunk 3]
```diff
@@ -177,21 +152,11 @@ export const HeaderMegaMenu: FC = () => {
...
-                  to={Paths.login}
-                  variant="default"
-                >
-                  {t('header.login')}
-                </Button>
-                <Button onClick={closeDrawer} component={Link} to={Paths.register}>
-                  {t('header.signup')}
-                </Button>
-              </>
+              <Button onClick={closeDrawer} component={Link} to={Paths.signin}>
```
Comment (remadyt) [inline src/components/Header/Header.tsx:155]: handleCloseDrawer

File: src/components/Header/Header.tsx [hunk 4]
```diff
@@ -131,8 +108,7 @@ export const HeaderMegaMenu: FC = () => {
                   src={null}
                   alt="no image here"
                 />
-                <Text fz="xl">{email}</Text>
-                <Button onClick={handleLogOut}>
+                <Button onClick={signOut.signOut}>
```
Comment (remadyt) [inline src/components/Header/Header.tsx:111]: move to function and use destructuring, it is used in at least 2 places

File: src/components/SignInForm/SignInForm.tsx [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+} from '@mantine/core';
+import { useForm } from '@mantine/form';
+import { useTranslation } from 'react-i18next';
+
+import { LoaderScreen } from 'components';
+import { errorNotification, successNotification } from 'helpers';
+import { useStore } from 'store/store';
+
+export const SignInForm: FC<PaperProps> = (props) => {
+  const [showOTPInput, setShowOTPInput] = useState<boolean>(false);
```
Comment (remadyt) [inline src/components/SignInForm/SignInForm.tsx:21]: use 'is' prefix for boolean

File: src/components/SignInForm/SignInForm.tsx [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+  Stack,
+} from '@mantine/core';
+import { useForm } from '@mantine/form';
+import { useTranslation } from 'react-i18next';
+
+import { LoaderScreen } from 'components';
+import { errorNotification, successNotification } from 'helpers';
+import { useStore } from 'store/store';
+
+export const SignInForm: FC<PaperProps> = (props) => {
```
Comment (remadyt) [inline src/components/SignInForm/SignInForm.tsx:20]: use destructuring for props

File: src/components/SignInForm/SignInForm.tsx [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+      phone: '',
+    },
+
+    validate: {
+      phone: (value) =>
+        /^\+[1-9]\d{1,14}$/.test(value) ? null : 'Wrong phone number format!',
+    },
+  });
+
+  const OPTForm = useForm({
```
Comment (remadyt) [inline src/components/SignInForm/SignInForm.tsx:36]: why u use 2 forms?

File: src/components/SignInForm/SignInForm.tsx [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+  };
+
+  return (
+    <Paper radius="md" p="xl" withBorder {...props} pos="relative">
+      <Text size="lg" weight={500} ta="center">
+        {t('form.signin-header')}
+      </Text>
+      {showOTPInput ? (
+        <form onSubmit={OPTForm.onSubmit(handleOTP)}>
+          {signIn.isFetching && <LoaderScreen />}
```
Comment (remadyt) [inline src/components/SignInForm/SignInForm.tsx:80]: use desctructuring

File: src/components/SignInForm/SignInForm.tsx [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+    if (error) {
+      const title = t('notifications.error-header');
+      const message = t('notifications.verification-error');
+
+      errorNotification(title, message);
+    } else {
+      const title = t('notifications.success-header');
+      const message = t('notifications.signin-success');
+
+      successNotification(title, message);
```
Comment (remadyt) [inline src/components/SignInForm/SignInForm.tsx:55]: do not use if else
```
    let title = t('notifications.error-header');
    let message = t('notifications.verification-error');
if(error) {
      errorNotification(title, message);
      
      return;
}

      title = t('notifications.error-header');
      message = t('notifications.verification-error');

      errorNotification(title, message);

File: src/components/SignInForm/SignInForm.tsx [hunk 10]
```diff
@@ -0,0 +1,130 @@
...
+              </Button>
+              <Button type="submit" radius="xl">
+                {t('form.send')}
+              </Button>
+            </>
+          </Group>
+        </form>
+      ) : (
+        <form onSubmit={phoneForm.onSubmit(handlePhone)}>
+          {getOTP.isFetching && <LoaderScreen />}
```
Comment (remadyt) [inline src/components/SignInForm/SignInForm.tsx:107]: use container here

File: src/integrations/firebase/phoneAuth.ts [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+
+type CustomWindow = {
+  recaptchaVerifier: RecaptchaVerifier;
+  confirmationResult: ConfirmationResult;
+};
+
+type FBWindow = typeof window & CustomWindow;
+
+export const generateRecaptcha = (): void => {
+  (window as FBWindow).recaptchaVerifier = new RecaptchaVerifier(
```
Comment (remadyt) [inline src/integrations/firebase/phoneAuth.ts:19]: need move this to const
```
const extendedWindow = (window as FBWindow);
and reuse in all palces
```

File: src/integrations/firebase/phoneAuth.ts [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+
+export const appGetOTP = async (phone: string): Promise<void> => {
+  const appVerifier = (window as FBWindow).recaptchaVerifier;
+  const confirmationResult = await signInWithPhoneNumber(auth, phone, appVerifier);
+
+  (window as FBWindow).confirmationResult = confirmationResult;
+};
+
+export const appSignIn = async (code: string): Promise<User> => {
+  const result = await (window as FBWindow).confirmationResult.confirm(code);
```
Comment (remadyt) [inline src/integrations/firebase/phoneAuth.ts:36]: const { user }

File: src/integrations/firebase/phoneAuth.ts [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+  const { user } = result;
+
+  return user;
+};
+
+export const appSignOut = async (): Promise<void> => {
```
Comment (remadyt) [inline src/integrations/firebase/phoneAuth.ts:42]: incorrect name

File: src/store/slices/initSlice/initSlice.ts [hunk 14]
```diff
@@ -0,0 +1,54 @@
+import { getAuth, onAuthStateChanged } from 'firebase/auth';
+
+import { BoundStore } from 'store/store';
+import { GenericStateCreator } from 'store/types';
+
+type AppUser = {
+  id: string;
+  phone: Nullable<string>;
```
Comment (remadyt) [inline src/store/slices/initSlice/initSlice.ts:8]: nullable is redundant here

File: src/store/slices/initSlice/initSlice.ts [hunk 15]
```diff
@@ -0,0 +1,54 @@
...
+  name: Nullable<string>;
+  image: Nullable<string>;
+};
+
+export type InitSlice = {
```
Comment (remadyt) [inline src/store/slices/initSlice/initSlice.ts:13]: incorrect name, IState

**Review by remadyt — CHANGES_REQUESTED** [review]

File: src/App.tsx (hunk 1 above)
Comment (thelastandrew) [reply src/App.tsx:19]: this in not a hook, it's a store 'action'

File: src/integrations/firebase/phoneAuth.ts (hunk 13 above)
Comment (thelastandrew) [reply src/integrations/firebase/phoneAuth.ts:41]: if rename to signOut, then it has conflict with the built-in firebase method

**Pull request merged**